### PR TITLE
Gaurav button component

### PR DESCRIPTION
**Pull request by gauravThakur-sgl**
Added typeScript types using `interface IButtonProps {}`
Added custom colors inside the tailwind css and used them
`colors: {
        'primary': '#1f499e',
        'destructive': '#ff0000',
        'success': '#16a34a',
        'soft': '#e0e7ff',
      }`
Improved the baseClass condition so that on using variant classes or size class two tailwind property dosen't duplicate.
Installed lucide-icons

### CODE REVIEW

File: gaurav/src/components/ui/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+function Button(props: any) {
+  const { title, type, variant = "default", size = "default", className, onClick, children }: IButtonProps = props;
+
+  const baseClasses = "inline-flex justify-center items-center shadow-md focus:outline-none whitespace-nowrap";
+  const sizeClasses = buttonSize[size] || buttonSize.default;
+  const variantClasses = buttonColors[variant] || buttonColors.default;
+
+  //className={`${variantClasses} ${baseClasses} ${sizeClasses} ${className || ''}
```
Comment (ujjwalSGL) [inline gaurav/src/components/ui/Button.tsx:49]: if not require remove comments

File: gaurav/src/components/ButtonUi.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        </div>
+        <div className="pt-6">
+          <div className="text-sm font-medium  rounded text-gray-600">Button with icon</div>
+          <Card className="flex justify-center items-center p-4 w-full gap-8" variant="link">
+            <Button title="mail xs" variant="withIcon" size="xs"><Mail className="h-4 w-4 pr-1" /></Button> 
+            <Button title="Send sm" variant="withIcon" size="sm"><Send className="h-5 w-5 pr-1" /></Button> 
+            <Button title="Inbox Default" variant="withIcon"><Inbox className="h-6 w-6 pr-2" /></Button> 
+            <Button title="Button lg" variant="withIcon" size="lg"><Layers3 className="h-7 w-7 pr-2" /></Button> 
+            <Button title="Button xl" variant="withIcon" size="xl" ><BrainCircuit className="h-8 w-8 pr-2" /></Button>
+            <Button title="+" variant="withIcon" size="icon" />
```
Comment (ujjwalSGL) [inline gaurav/src/components/ButtonUi.tsx:98]: Create custom iconClass in button component then use it as a prop

File: gaurav/src/components/ui/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  default: "h-9 px-4 py-2 font-medium rounded-md",
+  xs: "h-6 px-3 py-1 text-xs font-semibold rounded-md",
+  sm: "h-8 rounded-md px-4 py-2 text-sm font-medium rounded-md",
+  lg: "h-10 rounded-md px-6 py-3 text-lg font-medium rounded-md",
+  xl: "h-11 px-8 py-4 rounded-md w-full max-w-sm text-lg font-medium rounded-md",
+  icon: "h-10 w-10 p-1 rounded-full",
+};
+
+function Button(props: any) {
+  const { title, type, variant = "default", size = "default", className, onClick, children }: IButtonProps = props;
```
Comment (ujjwalSGL) [inline gaurav/src/components/ui/Button.tsx:43]: use Interface directly don't make it complex by assigning  props: any (also try not to use any)

**Review by ujjwalSGL — COMMENTED** [review]
Fix these changes

File: gaurav/src/components/ui/Button.tsx (hunk 1 above)
Comment (gauravThakur-sgl) [reply gaurav/src/components/ui/Button.tsx]: removed

**Review by ujjwalSGL — APPROVED** [review]

**Review by shubhamships — APPROVED** [review]

**Pull request merged**